geo/src/algorithm/affine_ops.rs

```diff
@@ -183,6 +187,37 @@ impl<T: CoordNum> AffineTransform<T> {
         )
     }
 
+    /// Return the inverse of a given transform. Composing a transform with its inverse yields
+    /// the [identity matrix](Self::identity)
+    pub fn inverse(&self) -> Option<Self>
+    where
+        <T as Neg>::Output: Mul<T>,
+        <<T as Neg>::Output as Mul<T>>::Output: ToPrimitive,
+    {
+        let a = self.0[0][0];
+        let b = self.0[0][1];
+        let xoff = self.0[0][2];
+        let d = self.0[1][0];
+        let e = self.0[1][1];
+        let yoff = self.0[1][2];
+
+        let determinant = a * e - b * d;
+
+        if determinant == T::zero() {
+            return None; // The matrix is not invertible
+        }
+
+        let inv_det = T::one() / determinant;
+        Some(Self::new(
+            e * inv_det,
+            T::from(-b * inv_det).unwrap(),
```

Comment:
These `T::from` are kind of unfortunate.

To get rid of them, we could constrain these newly added `Neg`'s to instead be `Neg<Output=T>`? 

Or do we actually want to support Neg w/ other Outputs? I don't think it really even occurred to me before that you could specify the output of Neg. I'm guessing it's for when you want to do something like:
```
// unsigned
impl Neg for MyUnsignedNumber {
    // output of negation can be negative, so output is signed
   type Output= MySignedNumber;
    ...
}
```

maybe that's why? I have no idea.

Anyway, if you'd rather keep it, it's NBD. I wouldn't be super surprised if it's all optimized away anyway. 😅